src/azul/terra.py

```diff
@@ -646,19 +646,26 @@ def for_registered_user(cls, authentication: OAuth2) -> 'TDRClient':
     def drs_client(self) -> DRSClient:
         return DRSClient(http_client=self._http_client)
 
-    def get_duos(self, source: TDRSourceRef) -> Optional[MutableJSON]:
+    def get_duos(self,
+                 source: TDRSourceRef
+                 ) -> tuple[str | None, MutableJSON | None]:
         response = self._retrieve_source(source)
         try:
             duos_id = response['duosFirecloudGroup']['duosId']
         except (KeyError, TypeError):
             log.warning('No DUOS ID available for %r', source.spec)
-            return None
+            return None, None
         else:
             url = self._duos_endpoint('dataset', 'registration', duos_id)
             response = self._request('GET', url)
             if response.status == 404:
                 log.warning('No DUOS dataset registration with ID %r from %r',
                             duos_id, source.spec)
-                return None
+                return None, None
             else:
-                return self._check_response(url, response)
+                response = self._check_response(url, response)
+                assert response['consentGroups']
+                for consent_group in response['consentGroups']:
```

Comment:
Can you please explain your reasoning as to why you would expect more than one consent group with the same `duos_id`?

Reply:
The consent group is returned as an list of dictionary. I was unaware we should only expect one inside this list.
```
{
    'studyId': 5838,
    'studyName': 'GREGoR UCSC Test',
    'studyType': 'COHORT_STUDY',
…
    'consentGroups': [
        {
            'datasetId': 1957,
            'datasetIdentifier': 'DUOS-000667',
            'consentGroupName': 'GREGoR Test Consent Group 3',
…
        }
    ]
}
```

Reply:
> I was unaware we should only expect one inside this list.

Who said we should? I wanted to hear your reasoning. It sounds like you're just guessing.

Reply:
My reasoning was the data structure. Since `consentGroups` is a list there can presumably be more than one dictionary within.

Reply:
Obviously, but based on what evidence did you write code that assumes that, if there are multiple entries, that all of them mention the same ID? It could also be that the response schema is reused in different endpoints and that in the endpoint we are using here, there can only be one entry, in which case we could just use `one()`. As I said, it seems that you are guessing. Have you looked at any documentation? Did you ask the support team for DUOS?

Reply:
Victoria from the Broad confirmed the `consentGroups` list should only contain one object, matching the requested DUOS ID (aka `datasetIdentifier`). https://ucsc-gi.slack.com/archives/C03TPJS54DC/p1733844501027369?thread_ts=1733784745.615579&cid=C03TPJS54DC